Patch from upstream developer Folkert van Heusden, to fix bind (-y) issue.
https://bugs.gentoo.org/show_bug.cgi?id=367537

--- httping-1.5.1/main.c	2011-07-27 23:33:34.000000000 +0400
+++ httping-1.5.1/main.c	2011-08-01 09:30:11.000000000 +0400
@@ -350,6 +350,7 @@
 					{
 						bind_to = (struct sockaddr_in *)&bind_to_6;
 						memset(&bind_to_6, 0x00, sizeof(bind_to_6));
+						bind_to_6.sin6_family = AF_INET6;
 
 						if (inet_pton(AF_INET6, optarg, &(bind_to_6.sin6_addr)) != 1)
 						{
@@ -360,6 +361,7 @@
 					{
 						bind_to = (struct sockaddr_in *)&bind_to_4;
 						memset(&bind_to_4, 0x00, sizeof(bind_to_4));
+						bind_to_4.sin_family = AF_INET;
 
 						if (inet_pton(AF_INET, optarg, &(bind_to_4.sin_addr)) != 1)
 						{
